azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachineScaleSet.java

```diff
@@ -0,0 +1,1037 @@
+             * inbound NAT pools in the load balancer.
+             * </p>
+             *
+             * @return the next stage of the virtual machine scale set update
+             */
+            WithApplicable withoutPrimaryInternalLoadBalancer();
+        }
+
+        /**
+         * Stage of the virtual machine scale set update allowing to remove association between the primary network interface
+         * configuration and backend of the load balancer.
+         */
+        interface WithoutPrimaryLoadBalancerBackend {
+            /**
+             * Removes association between the primary network interface configuration and backend of the internet facing
+             * load balancer.
+             *
+             * @param backendNames the existing backend names to remove
+             * @return the next stage of the virtual machine scale set update
+             */
+            WithApplicable withoutPrimaryInternetFacingLoadBalancerBackends(String ...backendNames);
+
+            /**
+             * Removes association between the primary network interface configuration and backend of the internal load balancer.
+             *
+             * @param backendNames the existing backend names to remove
+             * @return the next stage of the virtual machine scale set update
+             */
+            WithApplicable withoutPrimaryInternalLoadBalancerBackends(String ...backendNames);
+        }
+
+        /**
+         * Stage of the virtual machine scale set update allowing to remove association between the primary network interface
+         * configuration and inbound NAT pool of the load balancer.
+         */
+        interface WithoutPrimaryLoadBalancerNatPool {
+            /**
+             * Removes association between the primary network interface configuration and inbound NAT pool of the
+             * internet facing load balancer.
+             *
+             * @param natPoolNames the name of an existing inbound NAT pools to remove
+             * @return the next stage of the virtual machine scale set update
+             */
+            WithApplicable withoutPrimaryInternetFacingLoadBalancerNatPools(String ...natPoolNames);
+
+            /**
+             * Removes association between the primary network interface configuration and inbound NAT pool of the
+             * internal load balancer.
+             *
+             * @param natPoolNames the name of an existing inbound NAT pools to remove
+             * @return the next stage of the virtual machine scale set update
+             */
+            WithApplicable withoutPrimaryInternalLoadBalancerNatPools(String ...natPoolNames);
+        }
+
+        /**
+         * The stage of a virtual machine scale set update containing inputs for the resource to be updated
+         * (via {@link WithApplicable#apply()}).
+         */
+        interface WithApplicable extends
```

Comment:
Right, i mean WithApply.

The reason for this interface is - i want to maintain a context in the initial stages of update. i.e. during update i want show `withInternal|internetFacingLoadBalancer` and if user choose it then give him an option to specify the backends and natpool from the load balancer.